### DISCUSSION TITLE
Deployment by Tag ref to make the Tag/Release the subject on environments page

### DISCUSSION THREAD

**Discussion by gyurikol** [community/community#17292] [Actions]
When using the Github deployments API to CREATE deployment with reference to a tag, the deployment will feature other subjects like either a commit SHA or Pull-Request.

My proposition is that when the **ref** is a **Tag**, that the **Tag** or associated **Release** is made the subject on the environments/deployments page.

Example:
![Screen Shot 2022-05-26 at 13 44 05](https://user-images.githubusercontent.com/11619522/170413318-1b5916ff-b978-4b06-a80a-6d945550328a.png)

**Comment by yarinsa** [comment]
Would be very useful for us!

**Comment by benoittgt** [comment]
And deployment should be referenced under Release

**Reply by olivernybroe** [reply]
That would be very useful to also see it under the release when you deployed on a tag linked to that release

**Comment by olivernybroe** [comment]
We actually just spend a lot of time trying to figure out why it didn't show our tag. 

Seems like we are not the only ones who would love having this.

**Comment by andrewgosselin** [comment]
This would be such a good behavior!
I use teamcity for deployments which triggers when a new release/tag is created.
I also update the deployments in github based off of that build.

It would be awesome to be able to relate a tag to a deployment and see it in the release.

**Comment by joshft91** [comment]
I think this discussion here is the mentioning the same thing? https://github.com/orgs/community/discussions/39377

In either case I agree - I don't understand how it's useful to show the feature branch name and GitHub PR # of the last commit to go into the branch that is being deployed.

**Comment by piotrekkr** [comment]
I did not found your question and also asked about this around [a year ago](https://github.com/orgs/community/discussions/17598) but no answer so far too. Would be great to see actual ref used for deployment instead of some pull requests or commits you did not use when making deployment.

**Comment by joshft91** [comment]
It appears that the deployment history will either show the PR branch name/PR #, or just the short commit hash of a deployment trigger - is that correct? In the screenshot below, the top deployment was made from a feature branch (no PR opened against it) and it shows just the commit hash.

The 2nd (inactive) deployment was made from `main` and it references the merged in feature branch (FXXXX)/PR #. Are these the two views available for deployments?

![image](https://github.com/community/community/assets/1458101/ebd70d3b-ef41-4ec5-9c7f-25e89135ed36)

**Comment by Dogacel** [comment]
We have an on-demand workflow that deploys a given commit. Whenever we do that we see the latest merged PR in the `Deployments/History` tab. Therefore we never look at the deployment histories tab. We should be able to give a commit/tag to show in the deployments page.
